[CI] Merge docker images across platforms

giordano opened the pull request:
With the current setup, images for different platforms are pushed with the same tag, overriding each other.  With this change, instead, images built for different architectures are merged together.  I still need to do some more testing, but something like this should do the trick.

giordano commented on .github/workflows/Container.yml:
```diff
@@ -90,37 +109,99 @@ jobs:
-              type=raw,value=latest,enable=${{ matrix.default == true && (github.ref_type == 'tag' || (inputs.tag != '' && inputs.mark_as_latest)) }}
-              type=raw,value=dev,enable=${{ matrix.default == true && github.ref_type == 'branch' && inputs.tag == '' }}
-          labels: |
-              org.opencontainers.image.version=${{ steps.pkg.outputs.version }}
```
I honestly don't quite understand what was going on here.  If anything needs to be adapted, it should be done in the `tags` input in the `Build and push image` step below.

maleadt replied:
When forcing the workflow, there's params that can be set: 

<img width="337" height="344" alt="Image" src="https://github.com/user-attachments/assets/c255c599-c8d4-4946-882e-b9f20434f82c" />

The idea is to extract that here, and set that as metadata on the generated containers so that you can do `docker run ghcr.io/juliagpu/cuda.jl:dev` or `ghcr.io/juliagpu/cuda.jl:cuda-12` etc.

giordano commented:
See for example https://github.com/giordano/CUDA.jl/pkgs/container/cuda.jl/622086517?tag=dev-julia1.11-cuda12, where the tag has both architectures, instead of only the last one which happened to have been pushed.

maleadt committed:
[CI] Merge docker images across platforms

With the current setup, images for different platforms are pushed with
the same tag, overwriting each other. This refactors the workflow into
two jobs: a per-platform `build` job that pushes single-arch images by
digest (no tag) and a `merge` job that combines the digests into one
manifest list per variant.

The `dev` / `latest` / <version> alias tags are recreated in the merge
step for the default variant only.

[skip ci]

Co-Authored-By: Mosè Giordano <mose@gnu.org>
